src/theme/components/switch.ts

```diff
@@ -0,0 +1,76 @@
+import { switchAnatomy } from "@chakra-ui/anatomy";
+import { createMultiStyleConfigHelpers, defineStyle } from "@chakra-ui/react";
+
+const { definePartsStyle, defineMultiStyleConfig } =
+  createMultiStyleConfigHelpers(switchAnatomy.keys);
+
+const baseStyleContainer = defineStyle({});
+
+const baseStyleThumb = defineStyle({
+  borderRadius: "100px",
+  bg: "white",
+  boxShadow: "0px 4px 4px 0px rgba(0, 0, 0, 0.08)",
+  flexShrink: "0",
+});
+
+const baseStyleTrack = defineStyle({
+  bg: "gray.700",
+  alignItems: "center",
+  flexShrink: "0",
+  borderRadius: "100px",
+  _checked: {
+    bg: "primary.500",
+    justifyContent: "flex-end",
+  },
+  _disabled: {
+    bg: "gray.300",
+  },
+});
+
+// Combine the 3 components of the switch
+const baseStyle = definePartsStyle(() => ({
+  track: baseStyleTrack,
+  container: baseStyleContainer,
+  thumb: baseStyleThumb,
+}));
+
+const sizes = {
+  sm: definePartsStyle({
+    container: { width: "28px", height: "16px" },
+    track: {
+      width: "12px",
+      height: "12px",
+      padding: "2px 14px 2px 2px",
```

Comment:
Instead of adding padding to the track to move the "thumb", let's see if we can use `translateX` and transitions on the thumb itself like [Chakra's default theme](https://github.com/chakra-ui/chakra-ui/blob/main/packages/components/theme/src/components/switch.ts#L57C12-L57C12). I realize that code is a little hard to follow, so happy to pair on this.

Reply:
Do we need to use translateX?  The thumb appears to move appropriately due to the track's _checked { justifyContent: "flex-end", }

Reply:
using translate just gives it the animation that you [see in the docs](https://chakra-ui.com/docs/components/switch/usage). I pushed a commit to resolve that. Merging now